src/backend/distributed/planner/multi_explain.c

```diff
@@ -710,3 +715,73 @@ ExplainYAMLLineStarting(ExplainState *es)
 		appendStringInfoSpaces(es->str, es->indent * 2);
 	}
 }
+
+
+/*
+ * CoordinatorInsertSelectExplainScan is a custom scan explain callback function
+ * which is used to print explain information of a Citus plan for an INSERT INTO
+ * distributed_table SELECT ... query that is evaluated on the coordinator.
+ */
+void
+CoordinatorInsertSelectExplainScan(CustomScanState *node, List *ancestors,
+								   struct ExplainState *es)
+{
+	CitusScanState *scanState = (CitusScanState *) node;
+	MultiPlan *multiPlan = scanState->multiPlan;
+	Query *query = multiPlan->insertSelectSubquery;
+	IntoClause *into = NULL;
+	ParamListInfo params = NULL;
+	char *queryString = NULL;
+
+	if (es->analyze)
+	{
+		/* avoiding double execution here is tricky, error out for now */
+		ereport(ERROR, (errmsg("EXPLAIN ANALYZE is currently not supported for INSERT "
+							   "... SELECT commands via the coordinator")));
+	}
+
+	ExplainOpenGroup("Select Query", "Select Query", false, es);
+
+	/* explain the inner SELECT query */
+	CitusExplainOneQuery(query, into, es, queryString, params);
+
+	ExplainCloseGroup("Select Query", "Select Query", false, es);
+}
+
+
+/*
+ * CitusExplainOneQuery is simply a duplicate of ExplainOneQuery in explain.c, which
+ * is static.
+ */
+static void
+CitusExplainOneQuery(Query *query, IntoClause *into, ExplainState *es,
```

Comment:
Didn't mean primary motivation, was a side-effect, but I do like deleting copy-pasted code from PostgreSQL. Maybe we can look into what you suggest in a bit (for now I've simply made @marcocitus's changes version-aware in 42140bae37b903c29efae340bfbb06385e56c58e).